widget/diagramwidget/connectionpad.go

```diff
@@ -0,0 +1,348 @@
+package diagramwidget
+
+import (
+	"image/color"
+
+	"fyne.io/x/fyne/widget/diagramwidget/geometry/r2"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/driver/desktop"
+	"fyne.io/fyne/v2/widget"
+)
+
+const (
+	pointPadSize float32 = 10
+)
+
+// ConnectionPad is an interface to a connection shape on a DiagramElement.
+type ConnectionPad interface {
+	fyne.Widget
+	desktop.Hoverable
+	GetPadOwner() DiagramElement
+	GetCenterInDiagramCoordinates() fyne.Position
+	getConnectionPointInDiagramCoordinates(referencePoint fyne.Position) fyne.Position
+	MouseDown(*desktop.MouseEvent)
+	MouseUp(*desktop.MouseEvent)
+	SetPadColor(color.Color)
+}
+
+type connectionPad struct {
+	padOwner  DiagramElement
+	lineWidth float32
+	padColor  color.Color
+}
+
+func (cp *connectionPad) GetPadOwner() DiagramElement {
+	return cp.padOwner
+}
+
+// MouseDown responds to mouse down events
+func (pp *PointPad) MouseDown(event *desktop.MouseEvent) {
+	connectionTransaction := pp.padOwner.GetDiagram().ConnectionTransaction
+	if connectionTransaction != nil {
+		link := connectionTransaction.Link
+		if link.isConnectionAllowed(connectionTransaction.LinkPoint, pp) {
+			padOwnerPosition := pp.padOwner.Position()
+			pseudoEvent := &fyne.DragEvent{
+				PointEvent: fyne.PointEvent{},
+				Dragged:    fyne.NewDelta(event.Position.X+padOwnerPosition.X+10, event.Position.Y+padOwnerPosition.Y-10),
+			}
+			// the link point has to be changed before the handle is dragged
+			connectionTransaction.LinkPoint = connectionTransaction.Link.GetLinkPoints()[1]
+			link.GetHandle(TARGET.ToString()).Dragged(pseudoEvent)
+			link.Refresh()
+			link.SetSourcePad(pp)
+			link.Refresh()
+			link.GetDiagram().SelectDiagramElement(link)
+			link.ShowHandles()
+		}
+	}
+}
+
+// MouseDown responds to mouse down events
+func (rp *RectanglePad) MouseDown(event *desktop.MouseEvent) {
+	connectionTransaction := rp.padOwner.GetDiagram().ConnectionTransaction
+	if connectionTransaction != nil {
+		link := connectionTransaction.Link
+		if link.isConnectionAllowed(connectionTransaction.LinkPoint, rp) {
+			padOwnerPosition := rp.padOwner.Position()
+			pseudoEvent := &fyne.DragEvent{
+				PointEvent: fyne.PointEvent{},
```

Comment:
Default value. No need to set it here.

Reply:
I'm not sure what you are referencing here. Please clarify.

Reply:
You can remove the line just over my comment. You are setting a field to its default value.

Reply:
Got it. Done.